clients/src/test/java/org/apache/kafka/common/network/SelectorTest.java

```diff
@@ -127,8 +127,10 @@ public void testConnectionRefused() throws Exception {
         ServerSocket nonListeningSocket = new ServerSocket(0);
         int nonListeningPort = nonListeningSocket.getLocalPort();
         selector.connect(node, new InetSocketAddress("localhost", nonListeningPort), BUFFER_SIZE, BUFFER_SIZE);
-        while (selector.disconnected().contains(node))
+        while (selector.disconnected().containsKey(node)) {
+            assertEquals(ChannelState.NOT_CONNECTED, selector.disconnected().get(node));
```

Comment:
If the broker is misconfigured, we see a `Connection Refused`. This does go through the disconnect notification code path. In `NetworkClient#processDisconnection`, the new warning added for `NOT_CONNECTED` handles this case.
```

[main] DEBUG Connection with localhost/127.0.0.1 disconnected (org.apache.kafka.common.network.Selector:394)
java.net.ConnectException: Connection refused
	at sun.nio.ch.SocketChannelImpl.checkConnect(Native Method)
	at sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:717)
	at org.apache.kafka.common.network.PlaintextTransportLayer.finishConnect(PlaintextTransportLayer.java:50)
	at org.apache.kafka.common.network.KafkaChannel.finishConnect(KafkaChannel.java:95)
	at org.apache.kafka.common.network.Selector.pollSelectionKeys(Selector.java:354)
	at org.apache.kafka.common.network.Selector.poll(Selector.java:321)
	at org.apache.kafka.clients.NetworkClient.poll(NetworkClient.java:359)
	at org.apache.kafka.clients.consumer.internals.ConsumerNetworkClient.poll(ConsumerNetworkClient.java:230)
	at org.apache.kafka.clients.consumer.internals.ConsumerNetworkClient.poll(ConsumerNetworkClient.java:206)
	at org.apache.kafka.clients.consumer.internals.ConsumerNetworkClient.poll(ConsumerNetworkClient.java:197)
	at org.apache.kafka.clients.consumer.internals.ConsumerNetworkClient.awaitMetadataUpdate(ConsumerNetworkClient.java:132)
	at org.apache.kafka.clients.consumer.internals.AbstractCoordinator.ensureCoordinatorReady(AbstractCoordinator.java:223)
	at org.apache.kafka.clients.consumer.internals.AbstractCoordinator.ensureCoordinatorReady(AbstractCoordinator.java:200)
	at org.apache.kafka.clients.consumer.internals.ConsumerCoordinator.poll(ConsumerCoordinator.java:285)
	at org.apache.kafka.clients.consumer.KafkaConsumer.pollOnce(KafkaConsumer.java:1043)
	at org.apache.kafka.clients.consumer.KafkaConsumer.poll(KafkaConsumer.java:1008)
	at kafka.consumer.NewShinyConsumer.<init>(BaseConsumer.scala:58)
	at kafka.tools.ConsoleConsumer$.run(ConsoleConsumer.scala:72)
	at kafka.tools.ConsoleConsumer$.main(ConsoleConsumer.scala:53)
	at kafka.tools.ConsoleConsumer.main(ConsoleConsumer.scala)
[main] DEBUG Node -1 disconnected. (org.apache.kafka.clients.NetworkClient:605)
[main] WARN Connection to node -1 could not be established. Broker may not be available. (org.apache.kafka.clients.NetworkClient:496)
```